dot/node_integration_test.go

```diff
@@ -0,0 +1,329 @@
+func TestInitNode_LoadGenesisData(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genPath := NewTestGenesisAndRuntime(t)
+
+	cfg.Init.Genesis = genPath
+	cfg.Core.GrandpaAuthority = false
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	config := state.Config{
+		Path: cfg.Global.BasePath,
+	}
+	stateSrvc := state.NewService(config)
+
+	gen, err := genesis.NewGenesisFromJSONRaw(genPath)
+	require.NoError(t, err)
+
+	genTrie, err := genesis.NewTrieFromGenesis(gen)
+	require.NoError(t, err)
+
+	genesisHeader, err := types.NewHeader(common.NewHash([]byte{0}), genTrie.MustHash(), trie.EmptyHash,
+		big.NewInt(0), types.NewDigest())
+	require.NoError(t, err)
+
+	err = stateSrvc.Initialise(gen, genesisHeader, genTrie)
+	require.NoError(t, err)
+
+	err = stateSrvc.Start()
+	require.NoError(t, err)
+
+	defer func() {
+		err = stateSrvc.Stop()
+		require.NoError(t, err)
+	}()
+
+	gendata, err := stateSrvc.Base.LoadGenesisData()
+	require.NoError(t, err)
+
+	testGenesis := NewTestGenesis(t)
+
+	expected := &genesis.Data{
+		Name:       testGenesis.Name,
+		ID:         testGenesis.ID,
+		Bootnodes:  common.StringArrayToBytes(testGenesis.Bootnodes),
+		ProtocolID: testGenesis.ProtocolID,
+	}
+	require.Equal(t, expected, gendata)
+
+	genesisHeader, err = stateSrvc.Block.BestBlockHeader()
+	require.NoError(t, err)
+
+	stateRoot := genesisHeader.StateRoot
+	expectedHeader, err := types.NewHeader(common.NewHash([]byte{0}), stateRoot, trie.EmptyHash, big.NewInt(0),
+		types.NewDigest())
+	require.NoError(t, err)
+	require.Equal(t, expectedHeader.Hash(), genesisHeader.Hash())
+}
+
+func TestInitNode_LoadStorageRoot(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genPath := NewTestGenesisAndRuntime(t)
+
+	cfg.Core.Roles = types.FullNodeRole
+	cfg.Core.BabeAuthority = false
+	cfg.Core.GrandpaAuthority = false
+	cfg.Init.Genesis = genPath
+
+	gen, err := genesis.NewGenesisFromJSONRaw(genPath)
+	require.NoError(t, err)
+
+	err = InitNode(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	ed25519Keyring, _ := keystore.NewEd25519Keyring()
+	ks.Gran.Insert(ed25519Keyring.Alice())
+	sr25519Keyring, _ := keystore.NewSr25519Keyring()
+	ks.Babe.Insert(sr25519Keyring.Alice())
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	require.IsType(t, node, &Node{})
+
+	expected := &trie.Trie{}
+	err = expected.LoadFromMap(gen.GenesisFields().Raw["top"])
+	require.NoError(t, err)
+
+	expectedRoot, err := expected.Hash()
+	require.NoError(t, err)
+
+	mgr := node.Services.Get(&core.Service{})
+
+	coreSrvc, ok := mgr.(*core.Service)
+	require.True(t, ok, "could not find core service")
+	require.NotNil(t, coreSrvc)
+
+	stateRoot, err := coreSrvc.StorageRoot()
+	require.NoError(t, err)
+	require.Equal(t, expectedRoot, stateRoot)
+}
+
+func balanceKey(t *testing.T, publicKey [32]byte) (storageTrieKey []byte) {
+	accKey := append([]byte("balance:"), publicKey[:]...)
+	hash, err := common.Blake2bHash(accKey)
```

Comment:
nit `accountKey` would be nice explicitness-wise

```suggestion
	accountKey := append([]byte("balance:"), publicKey[:]...)
	hash, err := common.Blake2bHash(accountKey)
```

Reply:
Updated.